scripts/src/javatests/org/oppia/android/scripts/license/MavenDependenciesRetrieverTest.kt

```diff
@@ -0,0 +1,1312 @@
+      DEP_WITH_INVALID_LINKS,
+      DEP_WITH_DIRECT_LINK_ONLY_LICENSE
+    )
+    testBazelWorkspace.setUpWorkspaceForRulesJvmExternal(coordsList)
+    val thirdPartyPrefixCoordList = coordsList.map { coordinate ->
+      "//third_party:${omitVersionAndReplaceColonsPeriods(coordinate)}"
+    }
+    createThirdPartyAndroidBinary(thirdPartyPrefixCoordList)
+    writeThirdPartyBuildFile(coordsList)
+    val depsList = MavenDependenciesRetriever.retrieveThirdPartyMavenDependenciesList()
+    assertThat(depsList).contains(
+      omitVersionAndReplaceColonsHyphensPeriods(DEP_WITH_SCRAPABLE_LICENSE)
+    )
+    assertThat(depsList).contains(
+      omitVersionAndReplaceColonsHyphensPeriods(DEP_WITH_INVALID_LINKS)
+    )
+    assertThat(depsList).contains(
+      omitVersionAndReplaceColonsHyphensPeriods(
+        DEP_WITH_DIRECT_LINK_ONLY_LICENSE
+      )
+    )
+  }
+
+  @Test
+  fun testAddChangesFromTextProto_depsUpdated_returnsDepsListWithUpdatedDeps() {
+    val license1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+    }.build()
+    val updatedLicense1 = License.newBuilder().apply {
+      this.licenseName = "The Apache License, Version 2.0"
+      this.originalLink = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      this.scrapableLink = ScrapableLink.newBuilder().apply {
+        url = "https://www.apache.org/licenses/LICENSE-2.0.txt"
+      }.build()
+    }.build()
+    val license2 = License.newBuilder().apply {
+      this.licenseName = "Simplified BSD License"
+      this.originalLink = "https://www.opensource.org/licenses/bsd-license"
+      this.extractedCopyLink = ExtractedCopyLink.newBuilder().apply {
+        url = "https://local-copy/bsd-license"
+      }.build()
+    }.build()
+    val mavenDependenciesList = listOf(
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_SCRAPABLE_LICENSE
+        this.artifactVersion = DATA_BINDING_VERSION
+        this.addLicense(license1)
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_NO_LICENSE
+        this.artifactVersion = PROTO_LITE_VERSION
+        this.addAllLicense(listOf(license2))
+      }.build()
+    )
+    val updatedMavenDependenciesList = listOf(
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_SCRAPABLE_LICENSE
+        this.artifactVersion = DATA_BINDING_VERSION
+        this.addLicense(updatedLicense1)
+      }.build(),
+      MavenDependency.newBuilder().apply {
+        this.artifactName = DEP_WITH_NO_LICENSE
+        this.artifactVersion = PROTO_LITE_VERSION
+        this.addAllLicense(listOf(license2))
+      }.build()
+    )
+
+    val finalDepsList = MavenDependenciesRetriever.addChangesFromTextProto(
+      mavenDependenciesList,
+      updatedMavenDependenciesList
+    )
+    assertThat(finalDepsList.size).isEqualTo(2)
```

Comment:
```suggestion
    assertThat(finalDepsList).hasSize(2)
```

More idiomatic.

Ditto elsewhere.

Reply:
Done.